Soujanya2004/wanderlust-2024
# [Fix] : Refactor code with moving the routing logic from app.js to routes.js

### Description
This PR aims to improve code structure and maintainability by refactoring the routing logic from `app.js` to a dedicated `routes.js` file. This change adheres to modular coding principles, making it easier to manage and scale the codebase.
**This PR does the following**:
Moves all routing logic to routes.js to declutter app.js


### Related Issues
Fixes : #196 

### Changes

1. Created a new routes.js file to handle all routing logic separately.

2. Refactored app.js to import routes.js and removed inline route definitions.

### Testing Instructions
To test the changes:

- Pull this branch.
- Run npm install to ensure all dependencies are up-to-date.
- Run npm test to verify that the refactoring doesn’t break any existing functionality.
- Verify the following functionalities:
- All routes work as expected from the browser or API client.
- Any redirected routes still function correctly after the refactor.
- Flash messages and session handling work as intended.


### Additional Context
This PR refactors code structure based on GSSoC feedback to enhance readability and maintainability.

### Checklist
Make sure to check off all the items before submitting. Mark with [x] if done.
- [ ✔️] I have performed a self-review of my code
- [✔️ ] I have commented my code, particularly in hard-to-understand areas
- [ ✔️] My changes generate no new warnings
- [ ✔️] I am working on this issue under GSSOC
